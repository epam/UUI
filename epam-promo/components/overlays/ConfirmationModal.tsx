import * as React from 'react';
import { IModal } from '@epam/uui-core';
import { ModalBlocker, ModalWindow, FlexSpacer, ModalHeader, Button, ModalFooter, Panel, ScrollBars } from '../';
import { i18n } from '../../i18n';

export interface ConfirmationModalWindowProps extends IModal<any> {
    caption: any;
    bodyContent?: any;
    hideCancelButton?: boolean;
}

export class ConfirmationModal extends React.Component<ConfirmationModalWindowProps> {
    render() {
        let bodyContent;

        if (this.props.bodyContent) {
            bodyContent = <Panel margin="24">{this.props.bodyContent}</Panel>;
        }

        return (
            <ModalBlocker blockerShadow="dark" {...this.props}>
                <ModalWindow width="420">
                    <ModalHeader borderBottom title={this.props.caption} onClose={() => this.props.abort()} background="white" />
                    <ScrollBars>{bodyContent}</ScrollBars>
                    <ModalFooter background="white">
                        <FlexSpacer />
                        {this.props.hideCancelButton || (
                            <Button caption={i18n.form.modals.discardButton} onClick={() => this.props.success(false)} fill="white" color="gray50" />
                        )}
                        <Button caption={i18n.form.modals.saveButton} onClick={() => this.props.success(true)} color="green" />
                    </ModalFooter>
                </ModalWindow>
            </ModalBlocker>
        );
    }
}
