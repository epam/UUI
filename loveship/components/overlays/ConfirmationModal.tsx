import * as React from 'react';
import { IHasRawProps, IModal } from '@epam/uui-core';
import { ModalBlocker, ModalWindow, FlexSpacer, ModalHeader, Button, ModalFooter, Panel, ScrollBars } from '../';
import { i18n } from '../../i18n';

export interface ConfirmationModalWindowProps extends IModal<any>, IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
    caption: string;
    bodyContent?: any;
    hideCancelButton?: boolean;
    width?: '300' | '420' | '600' | '900';
}

export const ConfirmationModal = React.forwardRef<HTMLDivElement, ConfirmationModalWindowProps>((props, ref) => (
    <ModalBlocker ref={ref} blockerShadow="dark" {...props}>
        <ModalWindow width={props.width || '420'}>
            <ModalHeader borderBottom title={props.caption} onClose={() => props.abort()} background="white" />
            <ScrollBars>{props.bodyContent && <Panel margin="24">{props.bodyContent}</Panel>}</ScrollBars>
            <ModalFooter background="white" borderTop>
                <FlexSpacer />
                {props.hideCancelButton || (
                    <Button caption={i18n.confirmationModal.discardButton} onClick={() => props.success(false)} color="night400" />
                )}
                <Button caption={i18n.confirmationModal.saveButton} onClick={() => props.success(true)} color="grass" />
            </ModalFooter>
        </ModalWindow>
    </ModalBlocker>
));
