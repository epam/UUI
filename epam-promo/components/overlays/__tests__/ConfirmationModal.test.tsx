import React from 'react';
import { ConfirmationModal } from '../ConfirmationModal';
import renderer from 'react-test-renderer';

describe('ConfirmationModal', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<ConfirmationModal caption="Test" key="test-key" success={result => jest.fn(result)} abort={jest.fn} isActive zIndex={1} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', () => {
        const tree = renderer
            .create(
                <ConfirmationModal
                    caption="Test"
                    key="test-key"
                    success={result => jest.fn(result)}
                    abort={jest.fn}
                    isActive
                    zIndex={1}
                    bodyContent={<div>Test content</div>}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
