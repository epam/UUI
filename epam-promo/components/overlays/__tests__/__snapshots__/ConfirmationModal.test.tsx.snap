// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal should be rendered correctly 1`] = `
<div
  className="container modalBlocker shadow-dark"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="uui-modal-blocker"
    onClick={[Function]}
  />
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={1}
  />
  <div
    data-focus-lock-disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      aria-modal={true}
      className="uui-modal-window modal width-420 height-auto container"
      role="modal"
    >
      <div
        className="root size-36 background-white padding-24 vPadding-12 border-bottom-gray40 spacing-12 modalHeader container align-items-center"
      >
        <div
          className="root font-sans-semibold text-color-gray80 line-height-24 font-size-18 v-padding-12 container"
        >
          Test
        </div>
        <div
          className="flexSpacer"
        />
        <div
          className=""
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "auto",
              "flexGrow": undefined,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <button
            className="container uui-button-box uui-enabled -clickable root icon-color-gray60"
            href={null}
            onClick={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="root root theme-light"
        cx={
          Array [
            Array [
              "root",
              "theme-light",
            ],
            undefined,
          ]
        }
        style={
          Object {
            "display": "flex",
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "bottom": 0,
              "flex": "1 1 auto",
              "left": 0,
              "marginBottom": 0,
              "marginRight": 0,
              "overflow": "scroll",
              "position": "relative",
              "right": 0,
              "top": 0,
            }
          }
        />
        <div
          className="uui-track-horizontal"
          style={
            Object {
              "display": "none",
              "height": 6,
              "position": "absolute",
            }
          }
        >
          <div
            className="uui-thumb-horizontal"
          />
        </div>
        <div
          className="uui-track-vertical"
          style={
            Object {
              "display": "none",
              "position": "absolute",
              "width": 6,
            }
          }
        >
          <div
            className="uui-thumb-vertical"
          />
        </div>
      </div>
      <div
        className="root size-36 background-white padding-24 vPadding-24 spacing-12 modalFooter container align-items-center"
      >
        <div
          className="flexSpacer"
        />
        <button
          className="container uui-button-box uui-enabled -clickable button-color-gray50 root size-36 fill-white"
          href={null}
          onClick={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Discard
          </div>
        </button>
        <button
          className="container uui-button-box uui-enabled -clickable button-color-green root size-36 fill-solid"
          href={null}
          onClick={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Save
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
</div>
`;

exports[`ConfirmationModal should be rendered correctly 2`] = `
<div
  className="container modalBlocker shadow-dark"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="uui-modal-blocker"
    onClick={[Function]}
  />
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={1}
  />
  <div
    data-focus-lock-disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      aria-modal={true}
      className="uui-modal-window modal width-420 height-auto container"
      role="modal"
    >
      <div
        className="root size-36 background-white padding-24 vPadding-12 border-bottom-gray40 spacing-12 modalHeader container align-items-center"
      >
        <div
          className="root font-sans-semibold text-color-gray80 line-height-24 font-size-18 v-padding-12 container"
        >
          Test
        </div>
        <div
          className="flexSpacer"
        />
        <div
          className=""
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "auto",
              "flexGrow": undefined,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <button
            className="container uui-button-box uui-enabled -clickable root icon-color-gray60"
            href={null}
            onClick={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            <div
              className="container uui-icon uui-enabled"
              style={Object {}}
            >
              <svg
                className=""
              />
            </div>
          </button>
        </div>
      </div>
      <div
        className="root root theme-light"
        cx={
          Array [
            Array [
              "root",
              "theme-light",
            ],
            undefined,
          ]
        }
        style={
          Object {
            "display": "flex",
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "bottom": 0,
              "flex": "1 1 auto",
              "left": 0,
              "marginBottom": 0,
              "marginRight": 0,
              "overflow": "scroll",
              "position": "relative",
              "right": 0,
              "top": 0,
            }
          }
        >
          <div
            className="root margin-24 container"
          >
            <div>
              Test content
            </div>
          </div>
        </div>
        <div
          className="uui-track-horizontal"
          style={
            Object {
              "display": "none",
              "height": 6,
              "position": "absolute",
            }
          }
        >
          <div
            className="uui-thumb-horizontal"
          />
        </div>
        <div
          className="uui-track-vertical"
          style={
            Object {
              "display": "none",
              "position": "absolute",
              "width": 6,
            }
          }
        >
          <div
            className="uui-thumb-vertical"
          />
        </div>
      </div>
      <div
        className="root size-36 background-white padding-24 vPadding-24 spacing-12 modalFooter container align-items-center"
      >
        <div
          className="flexSpacer"
        />
        <button
          className="container uui-button-box uui-enabled -clickable button-color-gray50 root size-36 fill-white"
          href={null}
          onClick={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Discard
          </div>
        </button>
        <button
          className="container uui-button-box uui-enabled -clickable button-color-green root size-36 fill-solid"
          href={null}
          onClick={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            Save
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    data-focus-guard={true}
    style={
      Object {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={0}
  />
</div>
`;
